superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -115,18 +140,27 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
   };
 
   useEffect(() => {
-    handleChange(currentValue ?? []);
+    // For currentValue we need set always `FIRST_VALUE` only if we in config modal for `defaultToFirstItem` mode
+    handleChange(
+      isDefaultToFirstItemInFilterConfigModal
+        ? FIRST_VALUE
+        : currentValue ?? [],
+    );
   }, [
     JSON.stringify(currentValue),
+    defaultToFirstItem,
```

Comment:
Shouldn't this be `isDefaultToFirstItemInFilterConfigModal`?

Reply:
I think no,  because it's also relevant for a case when user had in filter bar not first value, and then in filter modal changed it to firstValue and clicked Save, so now it should be also updater in filter bar